### ISSUE TITLE
ENTSOE: 'Unknown' category for UK can be changed to biomass

### ISSUE THREAD

**Issue by FigRollLife** [tmrowco/electricitymap#219]
Awesome project, thanks!

I assume your UK data ultimately come from the national balancing mechanism service (bmreports.com). Your 'unknown' category appears to correspond to their 'other' category. If you download their BMU Fuel Type Excel Spreadsheet (go to Electricity Data Summary and scroll down to Generation by Fuel Type) you will see that 'other' comprises:
DRAXX-1
DRAXX-2
DRAXX-3
IRNPS-1
IRNPS-2
MARK-1
NSQYT-1
STCR-1
WILCT-1
WINN-1

The Drax units (DRAXX) are 100% biomass (wood pellets imported from North America) and contribute the vast majority of output within this category. We can ignore IRNPS (Ironbridge) as they closed down in 2015. MARK-1 is a wood chip CHP unit in Scotland. I'm not sure what the other plants are, but they seem to contribute negligible amounts of output.

So in summary I believe all the unknown generation is biomass, meaning you can switch your category name and CO2 outputs accordingly. I hope this helps.

Keep up the great work!

**Comment by brunolajoie** [comment]
Thanks a lot for this precious info, i'll have a look and change it accordingly! 
These are big numbers we are talking about so it matters

**Comment by corradio** [comment]
FYI the data comes from ENTSOE. I'm not sure where they get it from. It
might originate from bmreports though.

**Comment by ThierryOllivero** [comment]
ENTSOE is supposed to be fed by national TSO. For both capacity and generation.

**Comment by tezellis** [comment]
There's a definitive list of the major plants for the UK on page 146 here: https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/577712/DUKES_2016_FINAL.pdf

The only other one I could identify is
STCR-1  Stephen's Croft (Scotland)

**Comment by FigRollLife** [comment]
Great, thanks for chasing this up guys. Stephen's Croft is another wood-fired plant, so we're still looking at 100% wood so far.

**Comment by corradio** [comment]
Hi guys,

Because we fetch UK data through ENTSOE, I think it is better to contact them and have them correct their data rather than add a specific UK clause for biomass/other in our code here (this won't be scalable and might be error prone if they somehow change the way they segment data).

Is anybody up for contacting ENTSOE?

**Comment by brunolajoie** [comment]
Hello @corradio, I've double check and come to the same conclusion: "Unknown" category in ENTSOE data for UK seems to equal OTHER in bmreport.com. 

At the moment in our code, unknown carbon intensity is set to 300gCO2/kWh, while "Biomass" is set to 230 as per IPCC median value. Therefore, refining UK carbon intensity will entail only marginal changes after correction.

So, no high priority here, but I can still contact ENTSOE to see what we can do. Do you have any contact there?

**Comment by corradio** [comment]
Unfortunately not :(

**Comment by corradio** [comment]
cc @rustyb, are you able to help us?

**Comment by brunolajoie** [comment]
@corradio i think it's faire to change "unknown" production to "biomass". In order to help me settle this once for good and adjust the related "capacities.json" file, could you print us the maximum output per power plant type in GB that you registered in ENSTSO since we started to save data on our database? thanks.Bruno

**Comment by corradio** [comment]
@brunolajoie unfortunately it's not that easy. We would have to encode a special rule that says that UK unknown is actually biomass, and it would not be future proof (what happens if ENTSOE add something else to 'unknown'?)
I'd like to follow up with @rustyb from ENTSOE on this one. Maybe they can fix it. I renamed the issue to make it clearer.

**Comment by brunolajoie** [comment]
ok i got it! But, if you can still print me the max output you registered, that would help me fix the capacities.json. Else, no pb, will wait

**Comment by corradio** [comment]
Let's talk about it on slack

**Comment by brunolajoie** [comment]
I've updated the GB capacities with 2017 ENSTOE data.  There is now 2GW of biomass capacity and 6GW of unknown (2GW of unknown/other renewables, + 4GW of unknown/unknown). 

I suspect that ENTSOE real time production category "unknown" refers to the biomass capacity of 2GW, as we can see in the picture below, and should therefore be changed in our code to reflect this match.
![image](https://cloud.githubusercontent.com/assets/22095643/22373354/b514575c-e4a1-11e6-8806-7080942ee56f.png)

However, the 6GW of unknown category are really unknown to me, i don't see which actual output category could correspond to these 6GW of capacity.

**Comment by brunolajoie** [comment]
@rustyb any news on your side from ENTSOE for them to update their unknown category into biomass?

**Comment by rustyb** [comment]
Unfortunately I haven't heard anything back yet. Will post here when I do.

**Comment by corradio** [comment]
@rustyb friendly ping. Any news?

**Comment by otravers** [comment]
Note that only DRAXX-1/2/3 have so far been converted to biomass, their other units still run on coal for the time being.

**Comment by systemcatch** [comment]
Checking the BMU fuel type spreadsheet there are now only 2 units listed under other.  WINN-1 & NSQYT-1.

WINN-1 is the Winnington CHP plant in Cheshire, description [here](https://www.eonenergy.com/about-eon/our-company/generation/our-current-portfolio/combined-heat-and-power/about-us/winnington).

NSQYT-1 Despite my best efforts I can't find out what this plant is.  It started running early 2017 which means it should be fairly new and has the note EELPS which means "Embedded Exemptable
Large Power Station".  Best guess is solid waste or landfill gas, I might be able to find it in the DUKES digest.

Is ENTSOE showing things correctly now?

**Comment by brunolajoie** [comment]
it looks like ENTSOE data has been improved, not showing anymore unknwown since november, but biomass instead. If data is right, we can probably switch back unknwon to 700gCO2/kWh! What do you guy think?

![image](https://user-images.githubusercontent.com/22095643/34078468-4e6c0288-e31b-11e7-96ed-b7e2f4e4e3fd.png)

**Comment by corradio** [comment]
LGTM

**Comment by corradio** [comment]
@systemcatch any opinions? Anyone else? Else, @brunolajoie you can go ahead and make the changes.

**Comment by systemcatch** [comment]
Lets do it.
